packages/components/src/menu-item/types.d.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * External dependencies
+ */
+import type { ReactNode } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import type { IconType } from '../icon';
+
+export type MenuItemProps = {
+	/**
+	 * A CSS `class` to give to the container element.
+	 */
+	className?: string;
+	/**
+	 * The children elements.
+	 */
+	children?: ReactNode;
+	/**
+	 * Text to use as description for button text.
+	 */
+	info?: string;
+	/**
+	 * The icon to render. Supported values are: Dashicons (specified as
+	 * strings), functions, Component instances and `null`.
+	 *
+	 * @default null
+	 */
+	icon?: JSX.Element | null;
+	/**
+	 * Determines where to display the provided `icon`.
+	 */
+	iconPosition?: string;
```

Comment:
```suggestion
	iconPosition?: ButtonAsButtonProps[ 'iconPosition' ];
```

For this one specifically, I think it would be better to use the types from `Button` because it's stricter and more helpful.

Reply:
Sorted